auth/handlers/oauth2.go

```diff
@@ -0,0 +1,169 @@
+package handlers
+
+import (
+	"crypto"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/openfaas/openfaas-cloud/auth/provider"
+)
+
+// MakeOAuth2Handler makes a hanmdler for OAuth 2.0 redirects
+func MakeOAuth2Handler(config *Config) func(http.ResponseWriter, *http.Request) {
+	c := &http.Client{
+		Timeout: time.Second * 3,
+	}
+
+	privateKeydata, err := ioutil.ReadFile(config.PrivateKeyPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	privateKey, keyErr := jwt.ParseECPrivateKeyFromPEM(privateKeydata)
+	if keyErr != nil {
+		log.Fatal("Load private key", keyErr)
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		log.Printf(`OAuth 2 - "%s"`, r.URL.Path)
+		if r.URL.Path != "/oauth2/authorized" {
```

Comment:
Does `/oauth2/authorized/` need to be checked?

Reply:
I'm not following.